lib/Doctrine/ORM/Internal/CommitOrder/Edge.php

```diff
@@ -27,6 +29,13 @@ final class Edge
 
     public function __construct(string $from, string $to, int $weight)
     {
+        Deprecation::triggerIfCalledFromOutside(
+            'doctrine/orm',
+            'https://github.com/doctrine/orm/pull/10547',
+            'The %s class is deprecated and will be removed in ORM 3.0',
+            self::class
+        );
```

Comment:
If the class is deprecated, it should be annotated with `@deprecated`. That being said, it's already `@internal`, so we might as well remove it right away?

Reply:
I'd love to see it being removed right away, but the `CommitOrderCalculator` depends on it and was itself not marked as `@internal` (it only lived in a `...\Internal\...` namespace).

So, I'll keep the classes for now but added `@deprecated` annotations.